Packages/com.unity.inputsystem/Extras/NPad/NPadSupport.cs

```diff
@@ -0,0 +1,71 @@
+				{ ""name"" : ""StickL"", ""offset"" : 0, ""bit"" : 8 },
+				{ ""name"" : ""StickR"", ""offset"" : 0, ""bit"" : 9 },
+				{ ""name"" : ""L"", ""offset"" : 0, ""bit"" : 10 },
+				{ ""name"" : ""R"", ""offset"" : 0, ""bit"" : 11 },
+				{ ""name"" : ""ZL"", ""offset"" : 0, ""bit"" : 12 },
+				{ ""name"" : ""ZR"", ""offset"" : 0, ""bit"" : 13 },
+				{ ""name"" : ""Plus"", ""offset"" : 0, ""bit"" : 14 },
+				{ ""name"" : ""Minus"", ""offset"" : 0, ""bit"" : 15 },
+				{ ""name"" : ""LSL"", ""offset"" : 0, ""bit"" : 16 },
+				{ ""name"" : ""LSR"", ""offset"" : 0, ""bit"" : 17 },
+				{ ""name"" : ""RSL"", ""offset"" : 0, ""bit"" : 18 },
+				{ ""name"" : ""RSR"", ""offset"" : 0, ""bit"" : 19 },
+				{ ""name"" : ""VK_LUp"", ""offset"" : 0, ""bit"" : 20 },
+				{ ""name"" : ""VK_LDown"", ""offset"" : 0, ""bit"" : 21 },
+				{ ""name"" : ""VK_LLeft"", ""offset"" : 0, ""bit"" : 22 },
+				{ ""name"" : ""VK_LRight"", ""offset"" : 0, ""bit"" : 23 },
+				{ ""name"" : ""VK_RUp"", ""offset"" : 0, ""bit"" : 24 },
+				{ ""name"" : ""VK_RDown"", ""offset"" : 0, ""bit"" : 25 },
+				{ ""name"" : ""VK_RLeft"", ""offset"" : 0, ""bit"" : 26 },
+				{ ""name"" : ""VK_RRight"", ""offset"" : 0, ""bit"" : 27 },
+				{ ""name"" : ""leftStick"", ""offset"" : 4, ""format"" : ""VC2S"" },
+				{ ""name"" : ""leftStick/x"", ""offset"" : 0, ""format"" : ""SHRT"", ""parameters"" : ""clamp=false,invert=false,normalize=false"" },
+				{ ""name"" : ""leftStick/left"", ""offset"" : 0, ""format"" : ""SHRT"", ""parameters"" : ""invert=false,normalize=false"" },
+				{ ""name"" : ""leftStick/right"", ""offset"" : 0, ""format"" : ""SHRT"", ""parameters"" : ""invert=false,normalize=false"" },
+				{ ""name"" : ""leftStick/y"", ""offset"" : 2, ""format"" : ""SHRT"", ""parameters"" : ""clamp=false,invert=false,normalize=false"" },
+				{ ""name"" : ""leftStick/up"", ""offset"" : 2, ""format"" : ""SHRT"", ""parameters"" : ""invert=false,normalize=false"" },
+				{ ""name"" : ""leftStick/down"", ""offset"" : 2, ""format"" : ""SHRT"", ""parameters"" : ""invert=false,normalize=false"" },
+				{ ""name"" : ""rightStick"", ""offset"" : 8, ""format"" : ""VC2S"" },
+				{ ""name"" : ""rightStick/x"", ""offset"" : 0, ""format"" : ""SHRT"", ""parameters"" : ""clamp=false,invert=false,normalize=false"" },
+				{ ""name"" : ""rightStick/left"", ""offset"" : 0, ""format"" : ""SHRT"", ""parameters"" : ""invert=false,normalize=false"" },
+				{ ""name"" : ""rightStick/right"", ""offset"" : 0, ""format"" : ""SHRT"", ""parameters"" : ""invert=false,normalize=false"" },
+				{ ""name"" : ""rightStick/y"", ""offset"" : 2, ""format"" : ""SHRT"", ""parameters"" : ""clamp=false,invert=false,normalize=false"" },
+				{ ""name"" : ""rightStick/up"", ""offset"" : 2, ""format"" : ""SHRT"", ""parameters"" : ""invert=false,normalize=false"" },
+				{ ""name"" : ""rightStick/down"", ""offset"" : 2, ""format"" : ""SHRT"", ""parameters"" : ""invert=false,normalize=false"" }
```

Comment:
This seems to have the same state layout as the base template it is extending. What's the reason for having this template?

Reply:
I expect the template isn't derived any templates so I have to define all state layout. Isn't it?

Reply:
And, there are some Switch specific buttons so I'd like them to expose to developer.

Reply:
From what I see, it's extending NPad which takes its state layout from the state struct. That layout and the one in the JSON template here appear to be the same. Is the idea with NPad that we might end up supporting the Switch controller on platforms other than the Switch? I think I'm a little confused why we have two templates for what seems to be a single, platform-specific piece of hardware in my mind (a picture that may well be wrong).

Reply:
Yeah, documentation is "not great" ATM. I've started putting something together at https://github.com/Unity-Technologies/InputSystem/wiki/Input-Templates.

And no worries, I'm writing some tests for Npad and I'll bring it back together into a single template.